File: src/main/java/org/dedda/bratwurst/parse/StringGetCharParser.java
package org.dedda.bratwurst.parse;

import org.dedda.bratwurst.lang.BWExpression;
import org.dedda.bratwurst.lang.StringGetChar;

/**
 * Created by dedda on 1/25/16.
 *
 * @author dedda
 */
public class StringGetCharParser extends ExpressionParser {

    @Override
    public StringGetChar parse(String data, int lineNumber) {
        String[] split = data.split("\\}");
        String varName = split[0].substring(1);
        String location = split[1].substring(0, split[1].length() - 1);
        StringGetChar getChar = new StringGetChar(lineNumber, location, varName);
        return getChar;
    }
}

File: src/main/java/org/dedda/bratwurst/parse/StringGetCharParser.kt
package org.dedda.bratwurst.parse

import org.dedda.bratwurst.lang.StringGetChar

/**
 * Created by dedda on 1/25/16.
 *
 * @author dedda
 */
class StringGetCharParser : ExpressionParser() {
    override fun parse(data: String, lineNumber: Int): StringGetChar {
        val split = data.split("}".toRegex()).toTypedArray()
        val varName = split[0].substring(1)
        val location = split[1].substring(0, split[1].length - 1)
        return StringGetChar(lineNumber, location, varName)
    }
}